File: src/main/java/com/camackenzie/exvi/core/model/EnumUtils.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package com.camackenzie.exvi.core.model;

/**
 *
 * @author callum
 */
public class EnumUtils {

    public static <T extends Enum<T>> T enumFromString(Class<T> en, String st) {
        for (var enm : en.getEnumConstants()) {
            if (enm.toString().equalsIgnoreCase(st)) {
                return enm;
            }
        }
        throw new RuntimeException("Enum type could not be found froms string: " + st);
    }

    public static String formatName(String superStr) {
        return superStr.toLowerCase().replaceAll("_", " ");
    }

}

File: exvi-core/src/commonMain/kotlin/com/camackenzie/exvi/core/model/EnumUtils.kt
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package com.camackenzie.exvi.core.model


/**
 *
 * @author callum
 */
object EnumUtils {
    inline fun <reified T : Enum<T>> enumFromString(st: String): T? {
        for (enm in enumValues<T>()) {
            if (enm.toString().equals(st, ignoreCase = true)) {
                return enm
            }
        }
        return null
    }

    fun formatName(superStr: String): String {
        return superStr.lowercase().replace("_".toRegex(), " ")
    }
}